app/models/planning_application.rb

```diff
@@ -560,4 +569,16 @@ def determination_date_is_not_in_the_future
       errors.add(:determination_date, "Determination date must be today or in the past")
     end
   end
+
+  def set_application_number
+    self.application_number = local_authority.planning_applications.count + 100
```

Comment:
Would it be possible for 2 records initialised at the same time to end up with the same `application_number`? If so it's probably worth at least adding a validation to check it's unique.

I think there's a way to use PSQL to create an auto-incrementing column (https://www.geeksforgeeks.org/postgresql-create-auto-increment-column-using-serial/) but that's probably overkill!

Reply:
Ah, just seen the migration 😁